engine/execution/state/state.go

```diff
@@ -201,7 +201,19 @@ func RegisterValuesToValues(values []flow.RegisterValue) []ledger.Value {
 }
 
 func LedgerGetRegister(ldg ledger.Ledger, commitment flow.StateCommitment) delta.GetRegisterFunc {
+
+	readCache := make(map[flow.RegisterID]flow.RegisterEntry)
+
 	return func(owner, controller, key string) (flow.RegisterValue, error) {
+		regID := flow.RegisterID{
+			Owner:      owner,
+			Controller: controller,
+			Key:        key,
+		}
+
+		if value, ok := readCache[regID]; ok {
+			return value.Value, nil
+		}
```

Comment:
I moved the logic here, so it will be error-free and makes the view code cleaner